### ISSUE TITLE
Code release.

### ISSUE THREAD

**Issue by chenkang455** [GopiRajuMatta/BeSplat#1]
Hi @GopiRajuMatta where can i found the your paper code? thx a lot

**Comment by GopiRajuMatta** [comment]
Thank you for interest in our work.. Currently I am busy with paper submission.. I will refine the code base and release the code in a month.

**Comment by chenkang455** [comment]
> Thank you for interest in our work.. Currently I am busy with paper submission.. I will refine the code base and release the code in a month.

Get it, thx for your response and looking forward to your open-source.

**Comment by chenkang455** [comment]
Hi, @GopiRajuMatta. Sorry to bother you again. I was wondering if you have ever conducted ablation experiments on the event loss, the blur loss, and the combination of event and blur losses?

**Comment by GopiRajuMatta** [comment]
Yes, I've conducted but the results are not included in the paper.

Without event loss, as single-image deblurring problem is severely ill-posed, framework is not able to deblur.. 

Tried with various combinations of blur and event loss factors and included the results for which it works. For real dataset, I've observed it works with one set of blur and event loss factors, but for synthetic dataset, for each scene, I had to find out the suitable set of factors. Hope this answers your question. If anything, please feel to free to reach out.

Thank you
Gopi

**Comment by chenkang455** [comment]
> Yes, I've conducted but the results are not included in the paper.
> 
> Without event loss, as single-image deblurring problem is severely ill-posed, framework is not able to deblur.. 
> 
> Tried with various combinations of blur and event loss factors and included the results for which it works. For real dataset, I've observed it works with one set of blur and event loss factors, but for synthetic dataset, for each scene, I had to find out the suitable set of factors. Hope this answers your question. If anything, please feel to free to reach out.
> 
> Thank you
> Gopi

thx for your answer. i wonder do you have relative quantitative results? I have tried the combination of event and blur loss with weights 1 for each, but i found that the results are not satisfying. Specifically, event loss is sufficient for reconstructing 3d scenes and the addition of the blur loss brings little improvements. This has been bothering me for a period. Looking for your response. thx

**Comment by GopiRajuMatta** [comment]
Hope you are using normalized event loss which is mentioned in BeNeRF.. Also, which Gaussian Splatting baseline framework are you using?

**Comment by chenkang455** [comment]
> Hope you are using normalized event loss which is mentioned in BeNeRF.. Also, which Gaussian Splatting baseline framework are you using?

In the Tanabata scene, I employed the lin_log loss with a threshold c=0.1. My Gaussian framework was constructed based on the BAD-GS method. While I was able to successfully reconstruct the 3D scene using the event loss alone, I encountered difficulties when relying solely on the blur loss. I suspect this is due to the excessive blurriness of the images, which makes it challenging to decompose them effectively. Besides, when I attempted to directly combine the event and blur losses, the results were still not satisfactory.

So for the blur + event combination, it is better to use the normalized event loss instead of the normal? Below is my code:

```
def safe_log(x, eps=1e-9):
    return torch.log(x + eps)

reblur_3dgs = torch.mean(imgs_3dgs,dim = 0)
imgs_3dgs_log = safe_log(imgs_3dgs)
render_diff = imgs_3dgs_log[1:] - imgs_3dgs_log[:-1]
event_diff = event_voxel[..., None].repeat(1, 1, 1, 3) * 0.1
event_loss = torch.mean((render_diff - event_diff_norm) ** 2)

blur_loss = torch.mean((blur_image - reblur_3dgs) ** 2)
```

Looking forward to your reply! Thanks a lot!!

**Comment by GopiRajuMatta** [comment]
Once please check out BeNeRF, to get more clarity about event loss integration.. That you can find in their train.py file.. What do they do is instead of applying event loss on all consecutive pairs,  render_diff = imgs_3dgs_log[1:] - imgs_3dgs_log[:-1], they randomly sample only two rgb frames along the camera trajectory with 0.1 time difference, if you consider whole trajectory normalized to 0-1.. then they convert them to grayscale, then log difference, then loss will be calculated with respect to accumulated events over particular bin with same time difference at same time instances.. 

Also, it seems in above thing, you are not converting RGB frames to grayscale before giving them to log..

Please check their train.py once, you will get the idea. I incorporated the same into BeSplat framework, this worked. I used [Deblur-GS](https://github.com/Chaphlagical/Deblur-GS) framework instead of BAD-Gaussians..

If you have any other queries, please feel free..

Thank you
Gopi

**Comment by chenkang455** [comment]
> Once please check out BeNeRF, to get more clarity about event loss integration.. That you can find in their train.py file.. What do they do is instead of applying event loss on all consecutive pairs, render_diff = imgs_3dgs_log[1:] - imgs_3dgs_log[:-1], they randomly sample only two rgb frames along the camera trajectory with 0.1 time difference, if you consider whole trajectory normalized to 0-1.. then they convert them to grayscale, then log difference, then loss will be calculated with respect to accumulated events over particular bin with same time difference at same time instances..
> 
> Also, it seems in above thing, you are not converting RGB frames to grayscale before giving them to log..
> 
> Please check their train.py once, you will get the idea. I incorporated the same into BeSplat framework, this worked. I used [Deblur-GS](https://github.com/Chaphlagical/Deblur-GS) framework instead of BAD-Gaussians..
> 
> If you have any other queries, please feel free..
> 
> Thank you Gopi

Get it, thx for your reply!! Besides, I am curious about the performance gain of introducing blur loss based on the event loss, i.e., blur + event v.s. event .

**Comment by GopiRajuMatta** [comment]
In my case, just event loss didn't help, I had to use blur loss as well. Event loss is not giving right colors, only it is ensuring structure. So keeping blur loss is helping to right colors.

Thank you
Gopi

**Comment by chenkang455** [comment]
> In my case, just event loss didn't help, I had to use blur loss as well. Event loss is not giving right colors, only it is ensuring structure. So keeping blur loss is helping to right colors.
> 
> Thank you Gopi

thx your reply. I find that for the no-color scene, the event loss is enough and the blur loss brings no increment. I will investigate it further. Thank you!
